### ISSUE TITLE
[FR] general nll_loss and cross_entropy along arbitrary dimension

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#29754]
Currently it is always taken along `dim=1`. Making the dim configurable should be really easy. I'd do it if this sounds reasonable.

**Comment by vincentqb** [comment]
Sounds like a good idea to me.

@fmassa -- is your thumbs up meaning the same?

**Comment by vadimkantorov** [comment]
This would simplify the code doing views back and forth just to force cross_entropy to use last dimension without any contig copies
